File: packages/shared-types/src/task.ts
export type TaskStatus =
  | "queued"
  | "processing"
  | "completed"
  | "failed"
  | "cancelled";

export interface AnalysisTask {
  taskId: string;
  status: TaskStatus;
  progress: number;
  currentStep?: string;
  finishedVideos?: number;
  totalVideos?: number;
  error?: string;
}

export type TaskProgress = AnalysisTask;

File: apps/api/schemas/task.py
from pydantic import BaseModel


class UploadVideoResponse(BaseModel):
    videoId: str
    taskId: str
    status: str = "queued"


class TaskProgressResponse(BaseModel):
    taskId: str
    status: str
    progress: int = 0
    currentStep: str | None = None
    finishedVideos: int | None = None
    totalVideos: int | None = None
    error: str | None = None
    downloadUrl: str | None = None
